### PR TITLE
Chore: xcode12 update

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

minimal change to build with Xcode 12 on Azure

### TODO
- [x] before merging, restore `azure-pipelines.yml `
- [x] update snapshots with matching simulator
- [x] fix crashed/failed tests
- [x] ~~update carthage to 0.37~~ (wait for prebuild framework issue got fixed https://github.com/Carthage/Carthage/issues/3130)

### Dependencies:

step 0
- [x] ZipArchive
- [x] Cartography (4.0.0-xcode_12_4)
- [x] Down (v2.3.0_Xcode12.4)
- [x] ios-snapshot-test-case (4.0.0-xcode_12_4)
- [x] swift-snapshot-testing (1.8.2_Xcode12.4)
- [x] HTMLString (6.0.1_xcode12.4)
- [x] wireapp/PINCache (2.3-xcode12.4)
- [x] https://github.com/wireapp/wire-ios-shared-resources/pull/47

step 1
- [x] https://github.com/wireapp/wire-ios-canvas/pull/25
- [x]  https://github.com/wireapp/wire-ios-system/pull/55
- [x]  swift-protobuf (1.15.0_xcode12.4)

step 2
- [x] https://github.com/wireapp/wire-ios-testing/pull/34
- [x] https://github.com/wireapp/wire-ios-protos/pull/46

step 3
- [x] https://github.com/wireapp/wire-ios-utilities/pull/101

step 4
- [x] https://github.com/wireapp/wire-ios-link-preview/pull/72 (some tests disabled)
- [x] https://github.com/wireapp/wire-ios-images/pull/40
- [x] https://github.com/wireapp/wire-ios-cryptobox/pull/50
- [x] https://github.com/wireapp/wire-ios-transport/pull/176

step 5
- [x] https://github.com/wireapp/wire-ios-data-model/pull/1080 (1 test failed, applied tmp fix)
- [x] https://github.com/wireapp/wire-ios-mocktransport/pull/165 ~~(some tests disabled)~~

step 6
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/260/

step 7
- [x] https://github.com/wireapp/wire-ios-share-engine/pull/189/
- [x] https://github.com/wireapp/wire-ios-sync-engine/pull/1350
- [x] https://github.com/wireapp/wire-ios-ziphy/pull/29

regenerate snapshots and disable unstable tests
- [x] https://github.com/wireapp/wire-ios/pull/4956

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -9138,6 +9138,10 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = NO;
 				APPLICATION_EXTENSION_API_ONLY = NO;
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)/**",
+					"$(PROJECT_DIR)/Carthage/Build/iOS/**",
+				);
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9144]:
```suggestion

```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -9138,6 +9143,12 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = NO;
 				APPLICATION_EXTENSION_API_ONLY = NO;
+				EXCLUDED_ARCHS = "i386 arm64";
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(inherited)/**",
+					"$(PROJECT_DIR)/Carthage/Build/iOS/**",
+				);
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9150]:
```suggestion

```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -9088,6 +9086,7 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = NO;
 				APPLICATION_EXTENSION_API_ONLY = NO;
+				EXCLUDED_ARCHS = "i386 arm64";
```
Comment (typfel) [inline Wire-iOS.xcodeproj/project.pbxproj:9089]: Shouldn't this go into the [Base.xcconfig](https://github.com/wireapp/wire-ios/blob/develop/Wire-iOS/Resources/Configuration/Project-Base.xcconfig)

Same thing applies for the other build settings changes.

File: carthageXcode12.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+     
+
```
Comment (typfel) [inline carthageXcode12.sh:2]:
```suggestion
```

File: carthageXcode12.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#!/usr/bin/env bash
```
Comment (typfel) [inline carthageXcode12.sh:3]: So this scripts is currently duplicated in the `wire-ios-shared-resources` but the plan is to remove once the Carthage fix is out?

**Review by typfel — CHANGES_REQUESTED** [review]

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 3 above)
Comment (billypchan) [reply Wire-iOS.xcodeproj/project.pbxproj:9097]: It will be fixed in https://github.com/wireapp/wire-ios/pull/4963

File: carthageXcode12.sh (hunk 5 above)
Comment (billypchan) [reply carthageXcode12.sh:1]: yes, it is only need when building app, I will keep this script here til Carthage is fixed.

File: azure-pipelines.yml [hunk 6]
```diff
@@ -61,6 +61,6 @@ steps:
     fastlane post_test
   env:  
     CODECOV_TOKEN: $(CODECOV_TOKEN)
-    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/master"
+    DEPENDENCIES_BASE_URL: "https://raw.githubusercontent.com/wireapp/wire-ios-shared-resources/chore/xcode12"
```
Comment (johnxnguyen) [inline azure-pipelines.yml:64]: Does this need to point to the master branch?

File: carthageXcode12.sh (hunk 5 above)
Comment (johnxnguyen) [reply carthageXcode12.sh:1]: What is the issue with Carthage?

Comment (typfel) [reply carthageXcode12.sh:1]: The issue described [here](https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md) and and in the Carthage 0.37.0 [release notes](https://github.com/Carthage/Carthage/releases/tag/0.37.0)

tl;dr we must use a workaround until we migrate our frameworks to be `XCFrameworks`.

File: fastlane/Fastfile [hunk 7]
```diff
@@ -36,9 +37,19 @@ platform :ios do
             build = Build.new(options: options)
             # We need to update the AVS version before running setup script
             build.update_avs_version()
+
+            if cache_carthage == "Yes"
```
Comment (typfel) [inline fastlane/Fastfile:41]: Can you explain this Carthage cache option?

Comment (billypchan) [reply fastlane/Fastfile:41]: This is for speeding up building on Jenkins by reusing the Carthage folder from the last build, I will add some comments for it.

File: setup.sh [hunk 8]
```diff
@@ -46,15 +46,15 @@ XCODE_VERSION=( ${version//./ } )
...
+bash ./carthageXcode12.sh bootstrap --cache-builds --platform ios
 echo ""
 
 echo "ℹ️  Downloading AVS library..."
 ./Scripts/download-avs.sh
 echo ""
 
 echo "ℹ️  Downloading additional assets..."
-./Scripts/download-assets.sh "$@"
+./Scripts/download-assets.sh -b "chore/xcode12_simulator"
```
Comment (billypchan) [inline setup.sh:57]:
```suggestion
./Scripts/download-assets.sh "$@"
```

File: setup.sh [hunk 9]
```diff
@@ -46,15 +46,15 @@ XCODE_VERSION=( ${version//./ } )
...
+bash ./carthageXcode12.sh bootstrap --cache-builds --platform ios
 echo ""
 
 echo "ℹ️  Downloading AVS library..."
 ./Scripts/download-avs.sh
 echo ""
 
 echo "ℹ️  Downloading additional assets..."
-./Scripts/download-assets.sh "$@"
+./Scripts/download-assets.sh -b "fix/xcode12-debug-simulator"
```
Comment (billypchan) [inline setup.sh:57]:
```suggestion
./Scripts/download-assets.sh "$@"
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -9136,28 +9134,35 @@
 		8F42C556199244A700288E4D /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 i386";
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9137]:
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -9183,6 +9190,8 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = NO;
 				APPLICATION_EXTENSION_API_ONLY = NO;
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 i386";
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9192]:
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -9137,27 +9135,36 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				HEADER_SEARCH_PATHS = "$(SDKROOT)/usr/include/libxml2";
+				ONLY_ACTIVE_ARCH = YES;
+				SDKROOT = iphoneos;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9139]:
```suggestion
```

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -9137,27 +9135,36 @@
...
+				VALIDATE_WORKSPACE = YES;
 			};
 			name = Debug;
 		};
 		8F42C557199244A700288E4D /* Release */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				HEADER_SEARCH_PATHS = "$(SDKROOT)/usr/include/libxml2";
+				SDKROOT = iphoneos;
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/project.pbxproj:9148]:
```suggestion
```

File: setup.sh [hunk 14]
```diff
@@ -46,15 +46,15 @@ XCODE_VERSION=( ${version//./ } )
...
+bash ./carthageXcode12.sh bootstrap --cache-builds --platform ios
 echo ""
 
 echo "ℹ️  Downloading AVS library..."
 ./Scripts/download-avs.sh
 echo ""
 
 echo "ℹ️  Downloading additional assets..."
-./Scripts/download-assets.sh "$@"
+./Scripts/download-assets.sh -b fix/xcode12-debug-simulator
```
Comment (billypchan) [inline setup.sh:57]:
```suggestion
./Scripts/download-assets.sh "$@"
```

File: fastlane/Fastfile (hunk 7 above)
Comment (billypchan) [reply fastlane/Fastfile:46]: comment added

File: azure-pipelines.yml (hunk 6 above)
Comment (billypchan) [reply azure-pipelines.yml:64]: TODO added

**Review by johnxnguyen — APPROVED** [review]

**Review by marcomaddy1982 — APPROVED** [review]

**Comment by marcomaddy1982** [comment]
@billypchan Approved but some tests are failing.

**Comment by billypchan** [comment]
@marcomaddy1982 the snapshot updates are done here: https://github.com/wireapp/wire-ios/pull/4956. When that one is passed we can merge that PR to here.
